Source/NSPageController.m

```diff
@@ -0,0 +1,199 @@
+   version 2.1 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import <Foundation/NSArray.h>
+#import <Foundation/NSKeyValueObserving.h>
+#import <Foundation/NSArchiver.h>
+
+#import "AppKit/NSPageController.h"
+#import "AppKit/NSView.h"
+
+@implementation NSPageController
+
+- (instancetype) init
+{
+  self = [super init];
+  if (self != nil)
+    {
+      _transitionStyle = NSPageControllerTransitionStyleStackHistory;
+      _delegate = nil;
+      _arrangedObjects = [[NSMutableArray alloc] initWithCapacity: 10];
+      _selectedIndex = 0;
+      _selectedViewController = nil;
+    }
+  return self;
+}
+
+- (void) dealloc
+{
+  RELEASE(_arrangedObjects);
+  [super dealloc];
+}
+
+- (instancetype) initWithCoder: (NSCoder *)coder
+{
+  self = [super initWithCoder: coder];
+  if (self != nil)
+  {
+    if ([coder allowsKeyedCoding])
+      {
+        if ([coder containsValueForKey: @"NSTransitionStyle"])
+          {
+            _transitionStyle = [coder decodeIntForKey: @"NSTransitionStyle"];
+          }
+      }
+    else
+      {
+        [coder decodeValueOfObjCType: @encode(NSInteger)
+                                  at: &_transitionStyle];
+      }
+  }
+  return self;
+}
+
+- (void) encodeWithCoder: (NSCoder *)coder
+{
+  if ([coder allowsKeyedCoding])
+    {
+      [coder encodeInt: _transitionStyle
+                forKey: @"NSTransitionStyle"];
+    }
+  else
+    {
+      [coder encodeValueOfObjCType: @encode(NSInteger)
+                                at: &_transitionStyle];
+    }
+}
+
+// Set/Get properties
+- (NSPageControllerTransitionStyle) transitionStyle
+{
+  return _transitionStyle;
+}
+
+- (void) setTransitionStyle: (NSPageControllerTransitionStyle)style
+{
+  _transitionStyle = style;
+}
+
+- (id<NSPageControllerDelegate>) delegate
+{
+  return _delegate;
+}
+
+- (void) setDelegate: (id<NSPageControllerDelegate>)delegate
+{
+  _delegate = delegate;
+}
+
+- (NSArray *) arrangedObjects
+{
+  return _arrangedObjects;
+}
+
+- (void) setArrangedObjects: (NSArray *)array
+{
+  [_arrangedObjects removeAllObjects];
+  [_arrangedObjects addObjectsFromArray: array];
+}
+
+- (NSInteger) selectedIndex
+{
+  return _selectedIndex;
+}
+
+- (void) setSelectedIndex: (NSInteger)index
+{
+  if ([_delegate respondsToSelector: @selector(pageControllerWillStartLiveTransition:)])
+    {
+      [_delegate pageControllerWillStartLiveTransition: self];
+    }
+  [self willChangeValueForKey: @"selectedIndex"];
```

Comment:
These calls should happen automatically when the objects attribute gets observed.